server/apps/publish/archive_publish.py

```diff
@@ -342,12 +348,24 @@ def publish(self, doc, updates, target_media_type=None):
                                               if not any(a[config.ID_FIELD] == s[config.ID_FIELD] for s in subscribers)]
 
                 if len(subscribers_yet_to_receive) > 0:
+                    # Step 2(b)(iii)
+                    if updated.get('targeted_for'):
+                        subscribers_yet_to_receive = [s for s in subscribers_yet_to_receive
+                                                      if s.get('subscriber_type', '') != SUBSCRIBER_TYPES.DIGITAL]
+
+                    # Step 2(b)(iv)
                     formatters_not_found, queued_new_subscribers = \
                         self.queue_transmission(updated, subscribers_yet_to_receive, target_media_type)
                     no_formatters.extend(formatters_not_found)
         elif self.published_state == 'published':  # Step 2(c)
             # Step 2(c)(i)
             subscribers = list(get_resource_service('subscribers').get(req=None, lookup={'is_active': True}))
+
+            # Step 2(c)(ii)
+            if updated.get('targeted_for'):
+                subscribers = [s for s in subscribers if s.get('subscriber_type', '') != SUBSCRIBER_TYPES.DIGITAL]
```

Comment:
This filtering of subscribers that are not `DIGITAL` occurs in two places, perhaps we can create a function for that to avoid repeating the list comprehensions with identical logic, e.g.:
```py
from functools import partial
non_digital = partial(filter, lambda s: s.get('subscriber_type', '') != SUBSCRIBER_TYPES.DIGITAL)

...
subscribers_yet_to_receive = list(non_digital(subscribers_yet_to_receive))
...
subscribers = list(non_digital(subscribers))
```